tests/integration_tests/functional/test_cpu_template_helper.py

```diff
@@ -0,0 +1,264 @@
+            utils.run_cmd(
+                f"RUSTFLAGS='{cargo_build.get_rustflags()}' "
+                f"cargo build -p {self.BINARY_NAME} --release "
+                f"--target {cargo_build.DEFAULT_BUILD_TARGET}"
+            )
+            utils.run_cmd(f"strip --strip-debug {self.BINARY_PATH}")
+
+    def dump(self, vm_config_path, output_path):
+        """Dump CPU config in JSON format by calling dump subcommand"""
+        cmd = (
+            f"{self.BINARY_PATH} dump --config {vm_config_path} --output {output_path}"
+        )
+        utils.run_cmd(cmd)
+
+
+@pytest.fixture(scope="session", name="cpu_template_helper")
+def cpu_template_helper_fixture():
+    """Fixture of CPU template helper tool"""
+    return CpuTemplateHelper()
+
+
+def save_vm_config(microvm, vm_config_path):
+    """
+    Save VM config into JSON file.
+    """
+    config_json = microvm.full_cfg.get().json()
+    config_json["boot-source"]["kernel_image_path"] = str(microvm.kernel_file)
+    config_json["drives"][0]["path_on_host"] = str(microvm.rootfs_file)
+    with open(vm_config_path, "w", encoding="utf-8") as ofile:
+        ofile.write(json.dumps(config_json, indent=4))
+
+
+def build_cpu_config_dict(cpu_config_path):
+    """Build a dictionary from JSON CPU config file."""
+    cpu_config_dict = {
+        "cpuid": {},
+        "msrs": {},
+    }
+
+    with open(cpu_config_path, encoding="utf-8") as ifile:
+        cpu_config_json = json.loads(ifile.read())
+        # CPUID
+        for leaf_modifier in cpu_config_json["cpuid_modifiers"]:
+            for register_modifier in leaf_modifier["modifiers"]:
+                cpu_config_dict["cpuid"][
+                    (
+                        int(leaf_modifier["leaf"], 16),
+                        int(leaf_modifier["subleaf"], 16),
+                        register_modifier["register"],
+                    )
+                ] = int(register_modifier["bitmap"], 2)
+        # MSR
+        for msr_modifier in cpu_config_json["msr_modifiers"]:
+            cpu_config_dict["msrs"][int(msr_modifier["addr"], 16)] = int(
+                msr_modifier["bitmap"], 2
+            )
+
+    return cpu_config_dict
+
+
+CPUID_EXCEPTION_LIST = {
+    # CPUID.01h:ECX[OSXSAVE (bit 27)] is linked to CR4[OSXSAVE (bit 18)] that
+    # can be updated by guest OS.
+    # https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/arch/x86/kvm/x86.c?h=v5.10.176#n9872
+    (0x1, 0x0, "ecx"): 1 << 27,
+    # CPUID.07h:ECX[OSPKE (bit 4)] is linked to CR4[PKE (bit 22)] that can be
+    # updated by guest OS.
+    # https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/arch/x86/kvm/x86.c?h=v5.10.176#n9872
+    (0x7, 0x0, "ecx"): 1 << 4,
+    # CPUID.0Dh:EBX is variable depending on XCR0 that can be updated by guest
+    # OS with XSETBV instruction.
+    # https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/arch/x86/kvm/x86.c?h=v5.10.176#n973
+    (0xD, 0x0, "ebx"): 0xFFFFFFFF,
```

Comment:
nit: 0xFFFF_FFFF to make it easy on the eyes

Reply:
definitely, thanks!